components/brave_shields/browser/adblock_stub_response.cc

```diff
@@ -0,0 +1,113 @@
+    0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x4c, 0x01, 0x00, 0x3b};
+const unsigned char kJpeg1x1[] = {
+    0xff, 0xd8, 0xff, 0xdb, 0x00, 0x43, 0x00, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xc0, 0x00, 0x0b, 0x08, 0x00, 0x01, 0x00, 0x01, 0x01, 0x01, 0x11, 0x00,
+    0xff, 0xc4, 0x00, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0xff, 0xc4,
+    0x00, 0x14, 0x10, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xda, 0x00, 0x08,
+    0x01, 0x01, 0x00, 0x00, 0x3f, 0x00, 0x37, 0xff, 0xd9};
+
+// Basically, for now all Chromium image resource requests use hardcoded
+// 'Accept' header that starts with "image/webp". However, it is possible to
+// craft a custom 'Accept', for example, using XHR, so we provide stubs for
+// other popular mime types.
+std::string GetContentForMimeType(const std::string& mime_type) {
+  static const base::NoDestructor<
+      base::flat_map<std::string, std::string>>
+      content({
+          {"image/webp", {kWebp1x1, std::end(kWebp1x1)}},
+          {"image/*", {kPng1x1, std::end(kPng1x1)}},
+          {"image/apng", {kPng1x1, std::end(kPng1x1)}},
+          {"image/png", {kPng1x1, std::end(kPng1x1)}},
+          {"image/x-png", {kPng1x1, std::end(kPng1x1)}},
+          {"image/gif", {kGif1x1, std::end(kGif1x1)}},
+          {"image/jpeg", {kJpeg1x1, std::end(kJpeg1x1)}},
+      });
+  auto it = content->find(mime_type);
+  if (it == content->end()) {
+    return {};
+  }
+  return it->second;
+}
+
+}  // namespace
+
+void MakeStubResponse(const network::ResourceRequest& request,
+                      network::ResourceResponseHead* response,
+                      std::string* data) {
+  DCHECK(response);
+  DCHECK(data);
+
+  response->mime_type = "text/html";
+  *data = {};
+
+  // Possibly overwrite mime and stub data.
+  std::string accept_header;
+  request.headers.GetHeader("Accept", &accept_header);
+  auto mime_types = base::SplitString(
+      accept_header, ",;", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+  if (!mime_types.empty()) {
+    DCHECK(!mime_types.front().empty());
+    // If the entry looks like "*/*", use the default value. Otherwise, use
+    // the value from 'Accept', even if it looks like "audio/*".
+    if (mime_types.front()[0] != '*') {
+      response->mime_type = mime_types.front();
+    }
+    *data = GetContentForMimeType(response->mime_type);
+  }
+
+  // Craft response headers.
+  // TODO(iefremov): Allowing any origins still breaks some CORS requests.
```

Comment:
can you create an issue for this?

Reply:
There is one https://github.com/brave/brave-browser/issues/4396
Referenced it in the TODO